### PR TITLE
[JENKINS-70808] Fix isRestartable by only checking against the curren…

### PR DESCRIPTION
**Pull request by Dohbedoh**
…t node

# Description

See [JENKINS-70808](https://issues.jenkins.io/browse/JENKINS-70808).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by Dohbedoh** [comment]
cc @olamy

**Comment by olamy** [comment]
see https://github.com/jenkinsci/blueocean-plugin/pull/2414 this will store the result for future use and so will be faster

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java [hunk 1]
```diff
@@ -184,12 +192,48 @@ private WorkflowRun getRun() {
     @Override
     public boolean isRestartable() {
         RestartDeclarativePipelineAction restartDeclarativePipelineAction =
-            getRun().getAction( RestartDeclarativePipelineAction.class );
-        if (restartDeclarativePipelineAction != null) {
-            List<String> restartableStages = restartDeclarativePipelineAction.getRestartableStages();
-            if (restartableStages != null) {
-                return restartableStages.contains(this.getDisplayName())
-                    && this.getStateObj() == BlueRun.BlueRunState.FINISHED;
+            this.run.getAction( RestartDeclarativePipelineAction.class );
```
Comment (olamy) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java:195]: this.run cannot work anymore here need to use `getRun()`

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java [hunk 2]
```diff
@@ -184,12 +192,48 @@ private WorkflowRun getRun() {
...
+        return false;
+    }
+
+    /**
+     * @return the Pipeline execution
+     * @todo: This can be removed once https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/596 is available.
+     */
+    @edu.umd.cs.findbugs.annotations.CheckForNull
+    private CpsFlowExecution getExecution() {
+        FlowExecutionOwner owner = ((FlowExecutionOwner.Executable) run).asFlowExecutionOwner();
```
Comment (olamy) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java:209]: this.run cannot work anymore here need to use `getRun()`

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java [hunk 3]
```diff
@@ -184,12 +192,48 @@ private WorkflowRun getRun() {
...
+    /**
+     * Returns whether a stage is restartable.
+     * @param stageName the stage name
+     * @return true is restartable, false otherwise
+     * @todo: This can be removed once https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/596 is available.
+     */
+    private boolean isRestartable(String stageName) {
+        FlowExecution execution = getExecution();
+        if (execution != null) {
+            ExecutionModelAction execAction = run.getAction(ExecutionModelAction.class);
```
Comment (olamy) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java:226]: this.run cannot work anymore here need to use `getRun()`

**Review by olamy — CHANGES_REQUESTED** [review]

**Comment by olamy** [comment]
definitely computation need to be done in the constructor as the PR #2414 
because of this https://github.com/jenkinsci/blueocean-plugin/blob/95279d395af65b0ce545396dd530107873134a87/blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeContainerImpl.java#L35 
until the job is finished `PipelineNodeContainerImpl` objects are rebuild.

**Comment by olamy** [comment]
would be good to delete this PR and just finish this one first https://github.com/jenkinsci/blueocean-plugin/pull/2416 
then we merge it to master

**Comment by olamy** [comment]
done via cherrrypick of the commit from the branch https://github.com/jenkinsci/blueocean-plugin/pull/2417

**Comment by olamy** [comment]
done via https://github.com/jenkinsci/blueocean-plugin/commit/c284612ec2bcae03f247d24e5eab1ae535737ef0

**Pull request closed without merging**